dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryFactoryImpl.java

```diff
@@ -721,7 +721,13 @@ public void sortChildren(String inode)  throws DotDataException {
 			statement = conn.createStatement();
 			String sql;
 
-            PreparedStatement createSortPreparedStatement = conn.prepareStatement( catSQL.getCreateSortChildren() );
+            if ( DbConnectionFactory.isOracle() ){
+                //For Oracle we need to avoid ORA-01027 by creating the table before.
+                sql = catSQL.createCategoryReorderTable();

```

Comment:
As we talked in the hangouts: 
1. The logic to create-update-drop the category_reorder table has been always there, for Oracle I just divided it in two methods but follows the same pattern that have worked for us in the past.
2. I agree that this code is a bit of a mess but I don't feel comfortable changing it without completely understanding the original requirement.